pg_metadata/dock.py

```diff
@@ -314,3 +316,39 @@ def set_html_content(self, title=None, body=None):
     def default_html_content(self):
         self.set_html_content(
             'PgMetadata', tr('You should click on a layer in the legend which is stored in PostgreSQL.'))
+
+    @staticmethod
+    def set_layer_metadata(
+            layer: QgsVectorLayer,
+            connection: QgsAbstractDatabaseProviderConnection,
+            schema: str,
+            table: str,
+    ):
+        """ Set the main owner contact for the attribution. """
+        sql = (
+            "SELECT name FROM pgmetadata.v_contact "
+            f"WHERE table_name = '{table}' "
+            f"AND schema_name = '{schema}' "
+            "ORDER BY CASE "
+            "WHEN contact_role_code = 'OW' then 1 "
+            "WHEN contact_role_code = 'DI' then 2 "
+            "WHEN contact_role_code = 'CU' then 3 "
+            "ELSE 10 "
+            "END ASC "
+            "LIMIT 1;"
+        )
+
+        data = connection.executeSql(sql)
+        if not data:
+            return
+
+        if data[0] == NULL or data[0][0] == NULL:
+            return
+
+        # QGIS Server panel
+        layer.setAttribution(data[0][0])
```

Comment:
We could also add title and abstract to the QGIS Server corresponding properties ? Related to Lizmap plugin too

Reply:
which code can we use to add title and abstract to the QGIS server corresponding properties?

Reply:
@Mwekezi I think you should raise your topic in an issue, but not in a pull request. This one was a WIP, work in progress.
You need to use some Python to make it automatic for now. (semi-automatic to synchronize database metadata into the layer properties)

Reply:
Thank you. I tried running this code in the python console of QGIS but it didn't work. It mentioned an indentation error. Could you help with how I could successfully run the code above.

Reply:
Hi,
"Indentation error" is the most common Python error. Before digging in this project, I suggest you to read about programming with Python, follow a tutorial about Python, you will be able to understand why you couldn't run the code.